test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test 'Prevent sending reply without subject' do
+    login_as(@admin)
+    reply = @message.reply
+    reply.update_attributes(body: 'Reply Body', subject: nil)
+
+    MessageWorker.drain
+    assert msg = Message.last
+    assert_not_equal "Reply Body", msg.body
+  end
+
+  test 'Allow sending reply with subject' do
+    login_as(@admin)
+    reply = @message.reply
+    reply.update_attributes(body: 'Reply Body', subject: 'Reply Subject')
+
+    MessageWorker.drain
+    assert msg = Message.last
+    assert_equal "Reply Subject", msg.subject
```

Comment:
Same here, is it not more accurate to assert `message.valid?`

Reply:
Let me check this also and update test

Reply:
You're right, I misunderstood this test. I don't understand how checking the last record in the messages table gives us information about the `reply` message being delivered or not. Couldn't we use the `state` attribute to assert it? That would make the test easier to understand.